lib/class-wp-theme-json-gutenberg.php

```diff
@@ -2425,8 +2461,34 @@ function( $pseudo_selector ) use ( $selector ) {
 		$declarations_duotone = array();
 		foreach ( $declarations as $index => $declaration ) {
 			if ( 'filter' === $declaration['name'] ) {
-				unset( $declarations[ $index ] );
+				$filter_preset = $this->gutenberg_get_preset_from_declaration( $declarations[ $index ] );
```

Comment:
This is the one area that needs a lot more attention. What we have is a filter preset of the form `var(--wp--preset--duotone--(.*))`. What we need is a preset object from theme.json to get the settings so we can output the correct SVG. I think there must be a better way of getting the preset than using the slug like this.